Dockerfile: Use upstream wine-devel builds and ensure we always use it

3v1n0 opened the pull request:
This is enforced by setting the PATH to prefer the wine-devel build, so
that it's also used by winetricks

--- 

This doesn't fix #1 for me, but at least we get latest wine from a trusted source

3v1n0 committed:
Dockerfile: Use upstream wine-staging builds and ensure we always use it

This is enforced by setting the PATH to prefer the wine-devel build, so
that it's also used by winetricks

bingtimren commented:
Thanks for the pull request. However the image does not work for me. It shows up a blank GUI. Interestingly, the GUI still works. If I move the cursor to a supposedly button, it changes shape to indicate clickable. If I did click, it pops up a new window as supposed, although also blank.

![Screenshot from 2020-10-22 10-04-23](https://user-images.githubusercontent.com/2824513/96799235-60dc7f80-144e-11eb-9460-fcb2c4d1ee79.png)

3v1n0 commented:
Mmhmh, weird that you get #1 with this while it doesn't work for me the one based on wine-lol either :open_mouth:

3v1n0 committed:
Dockerfile: Disable GL acceleration as it depends on host settings

This may just cause black windows in most setups.
As per this, we can just disable the acceleration, it can be enabled by
launcher if possible in custom setups, but for generic one is just
better to disable it all the times.

Fixes #1

3v1n0 commented:
I've updated the image, with some changes... Now everything is working here.

The problem due to the black window is probably related to different driver combo in host, so using software rendering we avoid such problems. What gfx hardware do you have?

In any case I think it's safer to just disable the HW rendering for all, and maybe provide a setting in the launcher that unsets `LIBGL_ALWAYS_SOFTWARE` when provided.

![image](https://user-images.githubusercontent.com/345675/96896319-a8363080-148d-11eb-99a1-04e9adedd493.png)

bingtimren commented:
Bravo! It works perfectly. I'm now pushing the image built with your Dockerfile up to the repository.

I'm using a Lenovo E480 laptop. 

```
bing@bing-e480:~$ lspci
00:00.0 Host bridge: Intel Corporation Xeon E3-1200 v6/7th Gen Core Processor Host Bridge/DRAM Registers (rev 08)
00:02.0 VGA compatible controller: Intel Corporation UHD Graphics 620 (rev 07)
00:08.0 System peripheral: Intel Corporation Xeon E3-1200 v5/v6 / E3-1500 v5 / 6th/7th/8th Gen Core Processor Gaussian Mixture Model
00:14.0 USB controller: Intel Corporation Sunrise Point-LP USB 3.0 xHCI Controller (rev 21)
00:14.2 Signal processing controller: Intel Corporation Sunrise Point-LP Thermal subsystem (rev 21)
00:16.0 Communication controller: Intel Corporation Sunrise Point-LP CSME HECI #1 (rev 21)
00:17.0 SATA controller: Intel Corporation Sunrise Point-LP SATA Controller [AHCI mode] (rev 21)
00:1c.0 PCI bridge: Intel Corporation Sunrise Point-LP PCI Express Root Port #1 (rev f1)
00:1c.4 PCI bridge: Intel Corporation Sunrise Point-LP PCI Express Root Port #5 (rev f1)
00:1d.0 PCI bridge: Intel Corporation Sunrise Point-LP PCI Express Root Port #9 (rev f1)
00:1d.2 PCI bridge: Intel Corporation Sunrise Point-LP PCI Express Root Port #11 (rev f1)
00:1d.3 PCI bridge: Intel Corporation Device 9d1b (rev f1)
00:1f.0 ISA bridge: Intel Corporation Sunrise Point LPC Controller/eSPI Controller (rev 21)
00:1f.2 Memory controller: Intel Corporation Sunrise Point-LP PMC (rev 21)
00:1f.3 Audio device: Intel Corporation Sunrise Point-LP HD Audio (rev 21)
00:1f.4 SMBus: Intel Corporation Sunrise Point-LP SMBus (rev 21)
02:00.0 Display controller: Advanced Micro Devices, Inc. [AMD/ATI] Lexa PRO [Radeon 540/540X/550/550X / RX 540X/550/550X] (rev c0)
03:00.0 Ethernet controller: Realtek Semiconductor Co., Ltd. RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller (rev 10)
04:00.0 Non-Volatile memory controller: Samsung Electronics Co Ltd NVMe SSD Controller SM981/PM981/PM983
05:00.0 Network controller: Realtek Semiconductor Co., Ltd. RTL8822BE 802.11a/b/g/n/ac WiFi adapter
06:00.0 SD Host controller: O2 Micro, Inc. SD/MMC Card Reader Controller (rev 01)
bing@bing-e480:~$ 
```

And part of lshw output

```
           *-display
                description: Display controller
                product: Lexa PRO [Radeon 540/540X/550/550X / RX 540X/550/550X]
                vendor: Advanced Micro Devices, Inc. [AMD/ATI]
                physical id: 0
                bus info: pci@0000:02:00.0
                version: c0
                width: 64 bits
                clock: 33MHz
                capabilities: pm pciexpress msi bus_master cap_list rom
                configuration: driver=amdgpu latency=0
                resources: irq:150 memory:c0000000-cfffffff memory:d0000000-d01fffff ioport:d000(size=256) memory:f1400000-f143ffff memory:f1440000-f145ffff
```
